core/drivers/stm32_gpio.c

```diff
@@ -645,11 +654,29 @@ static int get_pinctrl_from_fdt(const void *fdt, int node,
 			ref->cfg.pupd = pull;
 			ref->cfg.od = odata;
 			ref->cfg.af = alternate;
+			ref->cfg.nsec = pin_non_secure;
+
+			bank_ref = stm32_gpio_get_bank(bank);
+
+			if (pin >= bank_ref->ngpios) {
+				EMSG("node %s requests pin %c%u that does not exist",
+				     fdt_get_name(fdt, consumer_node, NULL),
+				     bank + 'A', pin);
+				do_panic = true;
+			} else if (!pin_is_accessible(bank_ref, pin)) {
```

Comment:
`pin_is_accessible()` should also be checked when pinctrl is applied IMO. Same story as the secure state, it can be reconfigured in the meanswhile.

Reply:
This doesn't seem right. If the GPIO(s) are in semaphore, the semaphores will be taken whilst we don't apply the pinctrl, therefore don't consume it. I would remove this part. As I've said in my previous comment, if the CPU is not TDCID, we have no clue if the RIF configuration has been changed between getting the pinctrl and applying it.